### ISSUE TITLE
Accessing units

### ISSUE THREAD

**Issue by bjhardcastle** [AllenInstitute/npc_sessions#41]
- [x] upload and incorporate CCF annotations for available sessions
- [x] create capsule for exporting nwb to results asset or s3 directly
  - [ ] fix codeocean API request to run capsule for specified session (401 unauthorized error)
  - [ ] find a file format (hdf5/zarr) or loading method which speeds up access within codeocean
  - [ ] generate cross-session df with all units
- [ ] QC validate unit `peak_channel` <> CCF location @arjunsridhar12345
  - [ ] improve code making this join to ensure mix-ups cannot happen

**Comment by bjhardcastle** [comment]
To highlight one of the issues, when we access unit waveforms, the arrays don't match the expected number of channels:
```python 
>>> import npc_sessions
>>> s = npc_sessions.DynamicRoutingSession('644864_2023-02-02')
>>> {array.shape for array in s.units[:].waveform_mean}    
{(210, 378), (210, 334), (210, 358)}
``` 
- each array is `[time x channels]` so I expected that they should all be `(210 x 384)`
- it looks like waveform arrays are being saved with `sparsity=True` in `SpikeInterface`:
https://github.com/AllenNeuralDynamics/aind-capsule-ephys-spikesort-kilosort25-full/blob/101a25b11ab86388b9d79fe17d0899519c2f824a/code/run_capsule.py#L588-L596
..which might explain the `(210, 334)` array sizes, but doesn't explain the arrays with _more_ than 384 channels
  
@arjunsridhar12345 are you able to work on this soon? People are using the data, we need to be confident that it's correct - I don't think we fully understand what's going on at the moment...

**Comment by arjunsridhar12345** [comment]
yeah i can take a look. do you mean less? all of those examples have less than 384 channels. I think it could be when spikeinterface is discarding noisy/bad channels before calculating the waveforms

**Comment by arjunsridhar12345** [comment]
based on looking at the processing.json for this session, and seeing the number of noisy/bad channels removed and comparing that with the channel shape, that is the output after spikeinterface removes the noise/bad channels

**Comment by bjhardcastle** [comment]
> yeah i can take a look. do you mean less? all of those examples have less than 384 channels. I think it could be when spikeinterface is discarding noisy/bad channels before calculating the waveforms

Ha! Yes, they're all less than 384 - when I was exploring in the terminal I was thinking _348_ was the expected number of channels..

I think the correct way to deal with this is to populate each unit's `electrodes` value with the corresponding list of electrode IDs, not only the peak channel ID:
https://github.com/AllenInstitute/npc_sessions/blob/92d6458238f9b3b2a298541dfc6fe33f7caa3e63/src/npc_sessions/sessions.py#L1061C1-L1071C14

An assertion should be added to the effect of `all(unit.electrodes.size == unit.waveform_mean.shape[1] for unit in units)`
